### ISSUE TITLE
PagingController being prevented from rebuilding PagedListView

### ISSUE THREAD

**Issue by NateWilliams2** [EdsonBueno/infinite_scroll_pagination#28]
I've had a great time learning to use this plugin! 
After following your tutorial and documentation, I can't see what if anything I'm doing differently, but I'm having some issues. Within `_fetchPage`, right after `_pagingController.appendPage()` is called, I get the following exception: 
```bash
════════ Exception caught by foundation library ════════════════════════════════
The following assertion was thrown while dispatching notifications for PagingController<int, CardItem>:
setState() or markNeedsBuild() called during build.

This ValueListenableBuilder<PagingState<int, CardItem>> widget cannot be marked as needing to build because the framework is already in the process of building widgets.  A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was: ValueListenableBuilder<PagingState<int, CardItem>>
    state: _ValueListenableBuilderState<PagingState<int, CardItem>>#c3dcd
The widget which was currently being built when the offending call was made was: SliverList
    delegate: AppendedSliverChildBuilderDelegate#d62ee(estimated child count: 6)
```

The second page of items *is* indeed shown, but past the second page, the loading spinner is rendered despite `_nextPage` not even being called a third time. No more items are added or displayed beyond these first two pages, even though more exist. 

Is there something I'm getting wrong, or is this a bug? I'd appreciate help with this :) 

Relevant widget:
(replaced the `itemBuilder` with a simple text widget to simplify debugging)
```dart
class ContentListView extends StatefulWidget {
  @override
  _ContentListViewState createState() => _ContentListViewState();
}

class _ContentListViewState extends State<ContentListView> {
  static const _pageSize = 5;

  final PagingController<int, CardItem> _pagingController =
      PagingController(firstPageKey: 0, invisibleItemsThreshold: 2);

  @override
  void initState() {
    _pagingController.addPageRequestListener((pageKey) {
      _fetchPage(pageKey);
    });
    super.initState();
  }

  @override
  void didUpdateWidget(ContentListView oldWidget) {
    _pagingController.refresh();
    super.didUpdateWidget(oldWidget);
  }

  Future<void> _fetchPage(int pageKey) async {
    try {
      final newItems = List<CardItem>();
      for (var i = 0; i < _pageSize; i++) {
        newItems
            .add(CardItem.fromProcess(Globals.oldProcessFeed.getNextProcess()));
      }
      final isLastPage = !Globals.oldProcessFeed.hasNextItem;
      if (isLastPage) {
        _pagingController.appendLastPage(newItems);
      } else {
        final nextPageKey = pageKey + 1;
        _pagingController.appendPage(newItems, nextPageKey);
      }
    } catch (error) {
      _pagingController.error = error;
      log(error);
    }
  }

  @override
  Widget build(BuildContext context) => PagedListView<int, CardItem>(
        pagingController: _pagingController,
        builderDelegate: PagedChildBuilderDelegate<CardItem>(
          itemBuilder: (context, item, index) =>
              Text(item.process.processId).withTopPadding(300),
          newPageProgressIndicatorBuilder: (context) => Padding(
              padding: const EdgeInsets.only(
                top: 16,
                bottom: 16,
              ),
              child: Center(child: SpinnerCircular())),
        ),
      );

  @override
  void dispose() {
    _pagingController.dispose();
    super.dispose();
  }
}

```

**Comment by NateWilliams2** [comment]
Additionally, sometimes only the first page is displayed and the second page is never displayed at all.

**Comment by NateWilliams2** [comment]
Yep, sorry, should have mentioned I added that as a hopeful fix. Nothing changes when `didUpdateWidget` is there or not.

**Comment by NateWilliams2** [comment]
I should add, this app tree is nested within a `MaterialApp` and is contained by other `statefulWidget`s

**Comment by EdsonBueno** [comment]
I'll need a toy project to debug that. Can you provide one for me?

**Comment by EdsonBueno** [comment]
I'm closing this issue due to inactivity.
When you have your sample, please reach me on [our chat](https://gitter.im/infinite_scroll_pagination/community) and I'll be glad to help!

**Comment by EdsonBueno** [comment]
Thanks to your help on providing me a toy project, I was able to fix this issue in today's release (2.2.4)!
Thank you @NateWilliams2 !
